src/index.ts

```diff
@@ -415,10 +450,7 @@ function createParser(
       return jsonRootAst;
     }
 
-    let sortCompareFunction: (a: string, b: string) => number = lexicalSort;
-    if (jsonSortOrder) {
-      sortCompareFunction = createSortCompareFunction(jsonSortOrder);
-    }
+    const sortCompareFunction = createSortCompareFunction(jsonSortOrder);
```

Comment:
The default option of `{}` is now applied before this point, so we don't need to consider where `jsonSortOrder` is unset anymore.  `{}` is equivalent to lexical sort.